### DISCUSSION TITLE
What needs to be in: WEBTAU_FIREFOX_DRIVER_PATH

### DISCUSSION THREAD

**Discussion by aplatypus** [testingisdocumenting/webtau#1123] [Q&A] [answered]
* I do not have Chrome on my PC.  I do not need my life archived in Alphabet HQ.
* I have set `'WEBTAU_BROWSER_ID' => 'firefox'`
* WebTau didn't seem to try a download (as I expected)
* Instead it raised an exception:
  `org.openqa.selenium.WebDriverException: Cannot find firefox binary in PATH. Make sure firefox is installed.`
* Then I gave the path to the `firefox` binary
   `'/prod/apps/firefox/home/firefox'`
    * Again, same error: `org.openqa.selenium.WebDriverException: Cannot find firefox binary in PATH. Make sure firefox is installed.`

* Not sure which file or path to use for the Firefox browser option.
  * If there's something to download were do I find that??
  * Otherwise: which binary or path do I need to use for an installed Firefox?

**Comment by MykolaGolubyev** [comment, marked as answer]
Thanks for creating the topic. Will use this as a motivation to update docs.

There are two separate settings for Browser testing
* one is for browser binary itself, and most of the time its will use system installed executable (e.g. `firefoxBinPath` or `WEBTAU_FIREFOX_BIN_PATH` env var)
* one is for the selenium driver to control the browser and most of the time webtau autodownloads it. In case of firewall, users must download manually and set the path to the driver. (e.g. `firefoxDriverPath` or `WEBTAU_FIREFOX_DRIVER_PATH` env var)

**Reply by aplatypus** [reply]
* Thanks now I get it.
* `WEBTAU_FIREFOX_BIN_PATH` ............  is actually a file path to the Firefox executable.  Not a path to the location of the executable.
